src/model/ChillerElectricASHRAE205_Impl.hpp

```diff
@@ -0,0 +1,249 @@
+#define MODEL_CHILLERELECTRICASHRAE205_IMPL_HPP
+
+#include <model/ModelAPI.hpp>
+#include "WaterToWaterComponent_Impl.hpp"
+
+namespace openstudio {
+namespace model {
+
+  class ExternalFile;
+  class Node;
+  class ThermalZone;
+  class Schedule;
+
+  namespace detail {
+
+    /** ChillerElectricASHRAE205_Impl is a WaterToWaterComponent_Impl that is the implementation class for ChillerElectricASHRAE205.*/
+    class MODEL_API ChillerElectricASHRAE205_Impl : public WaterToWaterComponent_Impl
+    {
+     public:
+      /** @name Constructors and Destructors */
+      //@{
+
+      ChillerElectricASHRAE205_Impl(const IdfObject& idfObject, Model_Impl* model, bool keepHandle);
+
+      ChillerElectricASHRAE205_Impl(const openstudio::detail::WorkspaceObject_Impl& other, Model_Impl* model, bool keepHandle);
+
+      ChillerElectricASHRAE205_Impl(const ChillerElectricASHRAE205_Impl& other, Model_Impl* model, bool keepHandle);
+
+      virtual ~ChillerElectricASHRAE205_Impl() = default;
+
+      //@}
+      /** @name Virtual Methods */
+      //@{
+
+      virtual const std::vector<std::string>& outputVariableNames() const override;
+
+      virtual IddObjectType iddObjectType() const override;
+
+      virtual std::vector<ScheduleTypeKey> getScheduleTypeKeys(const Schedule& schedule) const override;
+
+      // chilledWaterLoop
+      virtual boost::optional<PlantLoop> plantLoop() const override;
+      virtual unsigned supplyInletPort() const override;
+      virtual unsigned supplyOutletPort() const override;
+
+      // condenserWaterLoop
+      virtual boost::optional<PlantLoop> secondaryPlantLoop() const override;
+      virtual unsigned demandInletPort() const override;
+      virtual unsigned demandOutletPort() const override;
+
+      // heatRecoveryLoop
+      virtual boost::optional<PlantLoop> tertiaryPlantLoop() const override;
+      virtual unsigned tertiaryInletPort() const override;
+      virtual unsigned tertiaryOutletPort() const override;
+
+      virtual std::vector<HVACComponent> edges(const boost::optional<HVACComponent>& prev) override;
+
+      /** This function will perform a check if trying to add it to a node that is on the demand side of a plant loop.
+       * If:
+       *     - the node is on the demand side of a loop
+       *     - the node isn't on the current condenser water loop itself
+       *     - the chiller doesn't already have a heat recovery (tertiary) loop,
+       * then it tries to add it to the Tertiary loop.
+       * In all other cases, it will call the base class' method WaterToWaterComponent_Impl::addToNode()
+       * If this is connecting to the demand side of a loop (not tertiary), will set the chiller condenserType to WaterCooled
+       */
+      virtual bool addToNode(Node& node) override;
+
+      /* Restricts addToTertiaryNode to a node that is on the demand side of a plant loop (tertiary = Heat Recovery Loop) */
+      virtual bool addToTertiaryNode(Node& node) override;
```

Comment:
I have to override pretty much everything due to the fact that this is the first object with more than 3 loops...